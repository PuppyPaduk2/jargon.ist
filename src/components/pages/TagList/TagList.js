// @flow

import React, { Fragment, PureComponent } from 'react';
import Link from 'gatsby-link';

import { Container, Section } from '../../ui';

import groupByFirstLetter, { type GroupedItem } from '../../../utils/groupByFirstLetter';
import createChunks from '../../../utils/chunks';

import { type Tag } from '../../../types/tag';

type Props = {
  tags: Array<Tag>,
};

type State = {
  chunks: Array<Array<GroupedItem<Tag>>>,
};

class TagList extends PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    const groupedJargons = groupByFirstLetter(props.tags, letter => letter.toLocaleUpperCase());

    this.state = {
      chunks: createChunks(groupedJargons, 4),
    };
  }

  render() {
    const { chunks } = this.state;

    return (
      <Fragment>
        <Section className="u-bg-steel-gray">
          <Container>
            <h1 className="u-color-primary u-clear-gap-top u-gap-bottom-small">Konular</h1>
            <p className="u-color-comet u-half-width@md-up u-full-width@md-down">
              Belli konular için gruplanmış jargonları görüntüleyin.
            </p>
          </Container>
        </Section>

        <Section className="u-pad-bottom-large">
          <Container>
            <div className="row">
              {chunks.map((chunk, index) => (
                /* eslint-disable-next-line react/no-array-index-key */
                <div key={index} className={`col col--lg-${12 / chunks.length}`}>
                  {chunk.map(group => (
                    <div key={group.letter} className="u-pad-bottom-large">
                      <h2 className="u-color-primary u-clear-gap-top">{group.letter}</h2>
                      {group.items.map(tag => (
                        <div key={tag.slug} className="u-gap-bottom-2xsmall">
                          <Link to={`/k/${tag.slug}`} className="c-link-secondary">
                            {tag.title}
                          </Link>
                        </div>
                      ))}
                    </div>
                  ))}
                </div>
              ))}
            </div>
          </Container>
        </Section>
      </Fragment>
    );
  }
}

export default TagList;
