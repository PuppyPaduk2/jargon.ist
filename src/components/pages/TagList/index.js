// @flow

import TagList from './TagList';

export default TagList;
