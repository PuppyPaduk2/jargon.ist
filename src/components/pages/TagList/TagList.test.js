// @flow

import React from 'react';
import { shallow } from 'enzyme';
import Link from 'gatsby-link';

import tagsFixture from '../../../../__fixtures__/tags';

import TagList from './TagList';

it('renders correctly', () => {
  const wrapper = shallow(<TagList tags={tagsFixture} />);

  expect(wrapper).toMatchSnapshot();
});

it('renders link to tag details', () => {
  const wrapper = shallow(<TagList tags={tagsFixture} />);

  tagsFixture.forEach(tag => {
    expect(wrapper.find(Link).find(`[to='/k/${tag.slug}']`)).toHaveLength(1);
  });
});
