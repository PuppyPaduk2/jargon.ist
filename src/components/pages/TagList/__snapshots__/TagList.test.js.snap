// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<React.Fragment>
  <h1>
    Konular
  </h1>
  <ul
    style={
      Object {
        "listStyle": "none",
        "margin": 0,
        "padding": 0,
      }
    }
  >
    <li
      className="u-gap-bottom-xsmall"
      key="tag1"
    >
      <GatsbyLink
        className="u-text-lowercase"
        to="/k/tag1"
      >
        tag1
      </GatsbyLink>
    </li>
    <li
      className="u-gap-bottom-xsmall"
      key="tag2"
    >
      <GatsbyLink
        className="u-text-lowercase"
        to="/k/tag2"
      >
        tag2
      </GatsbyLink>
    </li>
  </ul>
</React.Fragment>
`;
